users/views/settings/profile.py

```diff
@@ -50,14 +57,14 @@ def form_valid(self, form):
         icon = form.cleaned_data.get("icon")
         image = form.cleaned_data.get("image")
         if isinstance(icon, File):
-            resized_image = ImageOps.fit(Image.open(icon), (400, 400))
-            new_icon_bytes = io.BytesIO()
-            resized_image.save(new_icon_bytes, format=icon.format)
-            self.request.identity.icon.save(icon.name, File(new_icon_bytes))
+            self.request.identity.icon.save(
+                icon.name,
+                self.resize_image(icon, size=(400, 400)),
+            )
         if isinstance(image, File):
-            resized_image = ImageOps.fit(Image.open(image), (400, 400))
-            new_image_bytes = io.BytesIO()
-            resized_image.save(new_image_bytes, format=image.format)
-            self.request.identity.image.save(image.name, File(new_image_bytes))
+            self.request.identity.image.save(
+                image.name,
+                self.resize_image(image, size=(400, 400)),
```

Comment:
Just noticed I mangled the size here when I rewrote this yesterday - I'll fix it post merge.